### PR TITLE
update to 0.10.1

### PR DESCRIPTION
**Pull request by M-Waszkiewicz-Anaconda**
update composeml to 0.10.1 
 - [upstream](https://github.com/alteryx/compose/tree/v0.10.1)
- add private channel as there is cyclic dependency in tests

### CODE REVIEW

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,51 +1,62 @@
...
 source:
   url: https://pypi.io/packages/source/c/composeml/composeml-{{ version }}.tar.gz
-  sha256: 44a0707f7ae4cc482858fc4674931b082167d080c0c2c0a410fa85a022b02a6e
+  sha256: d87fc181be72fadec16ea0e44ebd363b4e2e3620c87c8b2cae6e236fd00761d4
 
 build:
-  noarch: python
   number: 0
-  script: {{ PYTHON }} -m pip install . --no-deps -vv
+  script: {{ PYTHON }} -m pip install --no-deps --no-build-isolation -vv .
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml:14]:
```suggestion
  skip: true  # [py<38]
  script: {{ PYTHON }} -m pip install --no-deps --no-build-isolation -vv .
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,51 +1,62 @@
...
-    - seaborn >=0.9.0
+    - python
+    - pandas >=1.3.0
+    - tqdm >=4.32.0
+    - matplotlib >=3.3.3
+    - seaborn >=0.11.0
 
 test:
+  source_files:
+    - composeml
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
```

File: recipe/meta.yaml [hunk 3]
```diff
@@ -1,51 +1,62 @@
...
   commands:
-    - pytest --pyargs composeml
+    - pip check
+    #skip failing tests connected with underlaying pandas issues eg  pandas._libs.tslibs.np_datetime.OutOfBoundsDatetime: Cannot cast 0001-01-01 02:00:00 to unit='ns' without overflow.
+    - pytest -k "not ( test_search_offset_mix_5 or test_minimum_data_per_group[minimum_data1] or test_sorted_distribution or test_subscriptable_slices[True-offsets2] or test_subscriptable_slices[True-offsets3])"
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - pytest --pyargs composeml -v -k "not ( test_search_offset_mix_5 or test_minimum_data_per_group[minimum_data1] or test_sorted_distribution or test_subscriptable_slices[True-offsets2] or test_subscriptable_slices[True-offsets3])"
```

File: recipe/meta.yaml [hunk 4]
```diff
@@ -1,51 +1,62 @@
...
-    - packaging ==20.4
-    - featuretools >=0.9.0
-  source_files:
-    - composeml/*
+    - pip
+    - pytest >=7.1.2
+    - pytest-xdist >=2.5.0
+    - featuretools >=1.27.0
+    - woodwork >=0.25.1
+    - pyarrow >=7.0.0
```
Comment (JeanChristopheMorinPerso) [inline recipe/meta.yaml]:
```suggestion
    - woodwork >=0.11.0
    - pyarrow >=3.0.0
```

**Review by JeanChristopheMorinPerso — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml [hunk 5]
```diff
@@ -1,51 +1,61 @@
...
-    - python >=3.6
-    - pandas >=0.23.0
-    - numpy >=1.13.3
-    - tqdm >=4.19.2
-    - matplotlib >=3.0.2
-    - seaborn >=0.9.0
+    - python
+    - pandas >=1.3.0
+    - tqdm >=4.32.0
+    - matplotlib >=3.3.3
```
Comment (skupr-anaconda) [inline recipe/meta.yaml]:
```suggestion
    - matplotlib-base >=3.3.3
```

**Comment by skupr-anaconda** [comment]
An error on `linux-s390x`:
```
conda.exceptions.ResolvePackageNotFound:
  - pyarrow[version='>=3.0.0']
```

An error on `osx-64`:
```
ChecksumMismatchError: Conda detected a mismatch between the expected content and downloaded content
for url 'https://conda.anaconda.org/marekw/osx-64/featuretools-1.28.0-py310h5e5eb69_0.conda'.
```

**Review by skupr-anaconda — APPROVED** [review]

**Review by cbouss — APPROVED** [review]
LGTM

**Pull request merged**